src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/BiosKeyboardInt9Handler.cs

```diff
@@ -11,8 +11,8 @@
 public class BiosKeyboardInt9Handler : InterruptHandler {
     private readonly Keyboard _keyboard;
 
-    public BiosKeyboardInt9Handler(Machine machine, ILoggerService loggerService) : base(machine, loggerService) {
-        _keyboard = machine.Keyboard;
+    public BiosKeyboardInt9Handler(Machine machine, Keyboard keyboard, ILoggerService loggerService) : base(machine, loggerService) {
```

Comment:
Yes, this is better, depend on the keyboard itself and not depend on the machine to provide a keyboard.
Now it just needs to depend on an IKeyboardDevice interface rather than a concrete implementation of the KeyBoard class.

Reply:
Done.